Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/lucene/impl/LuceneUtility.java

```diff
@@ -449,47 +455,77 @@ public void indexConcepts(String userName)
+                        queryString.append(searchBuffer.toString());
+                        if (!luceneFieldAnnotation.isTokenized()) {
+                            // If the field is not tokenzied, then the field
+                            // needs
+                            // to be analyzed using a whitespaceanalyzer, rather
+                            // than standard analyzer. This is because for non
+                            // tokenized strings we need exact matches and not
+                            // all
+                            // the nearest matches.
+                            analyzerPerField.put(luceneFieldAnnotation.lucenefieldName(), whiteSpaceAnalyzer);
                         }
                     }
-                    
-                    if (quoteOpen) {
-                        int idxLastQuote = searchBuffer.lastIndexOf("\"");
-                        searchBuffer.replace(idxLastQuote, idxLastQuote+1, "");
-                    }
-                    searchBuffer.append(")");
-                    queryString.append(searchBuffer.toString());
-                    if (!luceneFieldAnnotation.isTokenized()) {
-                        // If the field is not tokenzied, then the field needs
-                        // to be analyzed using a whitespaceanalyzer, rather
-                        // than standard analyzer. This is because for non
-                        // tokenized strings we need exact matches and not all
-                        // the nearest matches.
-                        analyzerPerField.put(luceneFieldAnnotation.lucenefieldName(), whiteSpaceAnalyzer);
+
+                } else {
+                    // Short words are allowed so index two times.
+                    // 1. Index with a String Field so that it is not tokenized
+                    // 2. Index with Text Field so that it is tokenized.
+                    if (searchString != null) {
+                        // Always analyze the short field name with whitespace
+                        // for non tokenized.
+                        analyzerPerField.put(luceneFieldAnnotation.luceneShortFieldName(), whiteSpaceAnalyzer);
+                        StringBuffer query = new StringBuffer("");
+                        query.append("(" + luceneFieldAnnotation.lucenefieldName()+":(+");
+                        query.append(searchString);
+                        query.append(" )");
+                        
+                        query.append(" OR ");
+                        
+                        // Second part of the query for short word. Exact match
+                        query.append(luceneFieldAnnotation.luceneShortFieldName() + ":");
+                        query.append("(+");
+                        query.append(searchString);
+                        query.append(" ))");
+
+                        queryString.append(query.toString());
+                        firstEntry++;
```

Comment:
what about exact matches (e.g. if someone tries to find "Albert Einstein" theory)? Also, have you tried using a QueryBuilder (https://lucene.apache.org/core/5_4_0/core/org/apache/lucene/util/QueryBuilder.html?is-external=true) and BooleanQueries, etc?